code/modules/atmospherics/machinery/components/unary_devices/thermomachine.dm

```diff
@@ -200,6 +221,17 @@
 	use_power(power_usage)
 	update_parents()
 
+/obj/machinery/atmospherics/components/unary/thermomachine/attackby(obj/item/assembly/signaler/anomaly/pyro, mob/living/user, params)
+	if(pyro_anomaly_core || !allows_cooling)
+		balloon_alert(user, "already an anomaly core!")
+		return
+	else
+		balloon_alert(user, "inserted anomaly core")
+		pyro.forceMove(src)
+		pyro_anomaly_core = pyro
+		RefreshParts()
+		return
```

Comment:
This return isn't needed, it's the end of the proc.

Or you can keep it so you can instead call parent below it for non-pyro anomaly things so they can work. Otherwise nothing can hit the thermomachine anymore.

Reply:
I removed the returns because I was mimicking what the other procs were doing. I am hesitantly keeping it so you can't hit the thermomachine because there are a variety of other things that behave that way such as station intercoms, posters, and newscasters. If this should be changed, please let me know.

Reply:
You should ALWAYS call parent. Even newscaster does.
![image](https://user-images.githubusercontent.com/53777086/236976078-29dedaf2-5679-4d70-835f-59bc6d4b35dc.png)
and poster
![image](https://user-images.githubusercontent.com/53777086/236976108-d31190f9-5103-459b-87c5-922ae392b0b5.png)

Intercom doesn't have an attackby at all so they just use parent.

Reply:
Gotcha, I added the parent call at the end, thank you.